.github/workflows/codespell.yml

```diff
@@ -0,0 +1,15 @@
+name: build
+
+on:
+  push:
```

Comment:
Need to restrict push to main branch. Otherwise, this will cause unnecessary CI runs for the forks.

Reply:
`main` and `release**`.

Reply:
This job is insanely quick so I think it’s fine and even beneficial to run it on every job. That way spelling errors can be catches before PRs are even openend.

Reply:
Most people (including me) don't usually check the tick mark of the CI in the commits in their own fork. I had done so, but rarely. Currently, the CI is run twice for each PR. Once, because it's a PR, twice, because it is pushed to the fork branch.

Reply:
I have no idea how other people develop, but I like to see as many parts of the CI pass as feasible before I open an PR. And running the code spell job has a very low cost, so I argue for keeping it this way.

Reply:
Updated the job so it only runs on pushes to main and release branches